### ISSUE TITLE
failed to sync remote state: No cluster leader

### ISSUE THREAD

**Issue by lkysow** [hashicorp/consul-helm#323]
_Originally posted by @yueru-mylove in https://github.com/hashicorp/consul-helm/issues/76#issuecomment-570472110_

The consul-0 starting logs is as belows:
>==> Log data will now stream in as it occurs:

    2020/01/03 01:57:14 [INFO]  raft: Initial configuration (index=0): []
    2020/01/03 01:57:14 [INFO] serf: EventMemberJoin: consul-0.dc1 10.244.0.102
    2020/01/03 01:57:14 [INFO]  raft: Node at 10.244.0.102:8300 [Follower] entering Follower state (Leader: "")
    2020/01/03 01:57:14 [INFO] serf: EventMemberJoin: consul-0 10.244.0.102
    2020/01/03 01:57:14 [INFO] agent: Started DNS server 0.0.0.0:8600 (udp)
    2020/01/03 01:57:14 [INFO] serf: Attempting re-join to previously known node: consul-2: 10.244.0.101:8301
    2020/01/03 01:57:14 [INFO] serf: Attempting re-join to previously known node: consul-2.dc1: 10.244.0.101:8302
    2020/01/03 01:57:14 [INFO] agent: Started DNS server 0.0.0.0:8600 (tcp)
    2020/01/03 01:57:14 [INFO] agent: Started HTTP server on [::]:8500 (tcp)
    2020/01/03 01:57:14 [INFO] consul: Adding LAN server consul-0 (Addr: tcp/10.244.0.102:8300) (DC: dc1)
    2020/01/03 01:57:14 [INFO] consul: Handled member-join event for server "consul-0.dc1" in area "wan"
    2020/01/03 01:57:14 [INFO] agent: Retry join LAN is supported for: aliyun aws azure digitalocean gce k8s mdns os packet scaleway softlayer triton vsphere
    2020/01/03 01:57:14 [INFO] agent: Joining LAN cluster...
    2020/01/03 01:57:14 [INFO] agent: (LAN) joining: [consul-0.consul.default.svc.cluster.local consul-1.consul.default.svc.cluster.local consul-2.consul.default.svc.cluster.local]
    2020/01/03 01:57:14 [INFO] agent: started state syncer
==> Consul agent running!
    2020/01/03 01:57:14 [WARN] memberlist: Failed to resolve consul-1.consul.default.svc.cluster.local: lookup consul-1.consul.default.svc.cluster.local on 10.10.0.10:53: no such host
    2020/01/03 01:57:14 [WARN] memberlist: Failed to resolve consul-2.consul.default.svc.cluster.local: lookup consul-2.consul.default.svc.cluster.local on 10.10.0.10:53: no such host
    2020/01/03 01:57:14 [INFO] agent: (LAN) joined: 1
    2020/01/03 01:57:14 [INFO] agent: Join LAN completed. Synced with 1 initial agents
    2020/01/03 01:57:17 [WARN] serf: Failed to re-join any previously known node
    2020/01/03 01:57:17 [WARN] serf: Failed to re-join any previously known node
    2020/01/03 01:57:21 [ERR] agent: failed to sync remote state: No cluster leader
    2020/01/03 01:57:22 [WARN]  raft: no known peers, aborting election
    2020/01/03 01:57:42 [ERR] agent: Coordinate update error: No cluster leader
    2020/01/03 01:57:54 [ERR] agent: failed to sync remote state: No cluster leader

And consul-1 and consul-2 only has log:
>==> Log data will now stream in as it occurs:

So I really don't  know why I got this. Consul service: 
```yaml
apiVersion: v1
kind: Service
metadata:
   name: consul
   labels:
     name: consul
spec:
   type: ClusterIP
   clusterIP: None
   ports:
    - name: http
      port: 8500
      targetPort: 8500
    - name: https
      port: 8443
      targetPort: 8443
    - name: rpc
      port: 8400
      targetPort: 8400
    - name: serflan-tcp
      protocol: "TCP"
      port: 8301
      targetPort: 8301
    - name: serflan-udp
      protocol: "UDP"
      port: 8301
      targetPort: 8301
    - name: serfwan-tcp
      protocol: "TCP"
      port: 8302
      targetPort: 8302
    - name: serfwan-udp
      protocol: "UDP"
      port: 8302
      targetPort: 8302
    - name: server
      port: 8300
      targetPort: 8300
    - name: consuldns
      port: 8600
      targetPort: 8600
   selector:
    app: consul
```

sts file: 
```yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: consul
spec:
  serviceName: consul
  replicas: 3
  selector:
    matchLabels:
      app: consul
  template:
    metadata:
      labels:
        app: consul
    spec:
      terminationGracePeriodSeconds: 10
      containers:
        - name: consul
          image: consul:latest
          args:
            - "agent"
            - "-server"
            - "-bootstrap-expect=3"
            - "-ui"
            - "-data-dir=/consul/data"
            - "-bind=0.0.0.0"
            - "-client=0.0.0.0"
            - "-advertise=$(PODIP)"
            - "-retry-join=consul-0.consul.$(NAMESPACE).svc.cluster.local"
            - "-retry-join=consul-1.consul.$(NAMESPACE).svc.cluster.local"
            - "-retry-join=consul-2.consul.$(NAMESPACE).svc.cluster.local"
            - "-domain=cluster.local"
            - "-disable-host-node-id"
          volumeMounts:
            - name: data
              mountPath: /consul/data
          env:
            - name: PODIP
              valueFrom:
                fieldRef:
                  fieldPath: status.podIP
            - name: NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
          ports:
            - containerPort: 8500
              name: ui-port
            - containerPort: 8400
              name: alt-port
            - containerPort: 53
              name: udp-port
            - containerPort: 8443
              name: https-port
            - containerPort: 8080
              name: http-port
            - containerPort: 8301
              name: serflan
            - containerPort: 8302
              name: serfwan
            - containerPort: 8600
              name: consuldns
            - containerPort: 8300
              name: server
      volumes:
        - name: data
          hostPath:
            path: /home/data
```

Thank you so much.

**Comment by lkysow** [comment]
@yueru-mylove it looks like you are not using the helm chart from this repository but using the one from `helm/charts`: https://github.com/helm/charts/tree/master/stable/consul.

We can't help you with that chart because it's not ours. You can either open an issue up in that repo or you can install our chart by following https://www.consul.io/docs/platform/k8s/run.html

**Comment by wiltonfelix** [comment]
Hello @lkysow  i had this problem:

my values.yaml:
```
# Available parameters and their default values for the Consul chart.

# global holds values that affect multiple components of the chart.
global:

  enabled: true

  # name sets the prefix used for all resources in the helm chart.
  # If not set, the prefix will be "<helm release name>-consul".
  name: null

  # domain is the domain Consul will answer DNS queries for
  # (see https://www.consul.io/docs/agent/options.html#_domain) and the domain
  # services synced from Consul into Kubernetes will have,
  # e.g. `service-name.service.consul`.
  domain: consul

  # image is the name (and tag) of the Consul Docker image for clients and
  # servers. This can be overridden per component.
  # This should be pinned to a specific version tag, otherwise you may
  # inadvertently upgrade your Consul version.
  #
  # Examples:
  #   # Consul 1.5.0
  #   image: "consul:1.5.0"
  #   # Consul Enterprise 1.5.0
  #   image: "hashicorp/consul-enterprise:1.5.0-ent"
  image: "consul:1.6.2"

  # imageK8S is the name (and tag) of the consul-k8s Docker image that
  # is used for functionality such as catalog sync. This can be overridden
  # per component.
  # Note: support for the catalog sync's liveness and readiness probes was added
  # to consul-k8s 0.6.0. If using an older consul-k8s version, you may need to
  # remove these checks to make the sync work.
  # If using bootstrapACLs then must be >= 0.10.1.
  # If using connect inject then must be >= 0.10.1.
  imageK8S: "hashicorp/consul-k8s:0.11.0"

  # datacenter is the name of the datacenter that the agents should register
  # as. This can't be changed once the Consul cluster is up and running
  # since Consul doesn't support an automatic way to change this value
  # currently: https://github.com/hashicorp/consul/issues/1858.
  datacenter: takidc

  # enablePodSecurityPolicies controls whether pod
  # security policies are created for the Consul components created by this
  # chart. See https://kubernetes.io/docs/concepts/policy/pod-security-policy/.
  enablePodSecurityPolicies: false

  # gossipEncryption configures which Kubernetes secret to retrieve Consul's
  # gossip encryption key from (see https://www.consul.io/docs/agent/options.html#_encrypt).
  # If secretName or secretKey are not set, gossip encryption will not be enabled.
  # The secret must be in the same namespace that Consul is installed into.
  #
  # The secret can be created by running:
  #    kubectl create secret generic consul-gossip-encryption-key \
  #      --from-literal=key=$(consul keygen).
  #
  # In this case, secretName would be "consul-gossip-encryption-key" and
  # secretKey would be "key".
  gossipEncryption:
    # secretName is the name of the Kubernetes secret that holds the gossip
    # encryption key. The secret must be in the same namespace that Consul is installed into.
    secretName: "consul"
    # secretKey is the key within the Kubernetes secret that holds the gossip
    # encryption key.
    secretKey: "gossip-encryption-key"

  # bootstrapACLs will automatically create and assign ACL tokens within
  # the Consul cluster. This requires servers to be running inside Kubernetes.
  # Additionally requires Consul >= 1.4 and consul-k8s >= 0.10.1.
  bootstrapACLs: false

  # Enables TLS encryption across the cluster to verify authenticity of the
  # servers and clients that connect. Note: It is HIGHLY recommended that you also
  # enable Gossip encryption.
  # See https://learn.hashicorp.com/consul/security-networking/agent-encryption
  #
  # Note: this relies on functionality introduced with Consul 1.4.1. Make sure
  # your global.image value is at least version 1.4.1.
  tls:
    enabled: true

    # serverAdditionalDNSSANs is a list of additional DNS names to
    # set as Subject Alternative Names (SANs) in the server certificate.
    # This is useful when you need to access the Consul server(s) externally,
    # for example, if you're using the UI.
    serverAdditionalDNSSANs: []

    # serverAdditionalIPSANs is a list of additional IP addresses to
    # set as Subject Alternative Names (SANs) in the server certificate.
    # This is useful when you need to access Consul server(s) externally,
    # for example, if you're using the UI.
    serverAdditionalIPSANs: []

    # If verify is true, 'verify_outgoing', 'verify_server_hostname', and
    # 'verify_incoming_rpc' will be set to true for Consul servers and clients.
    # Set this to false to incrementally roll out TLS on an existing Consul cluster.
    # Note: remember to switch it back to true once the rollout is complete.
    # Please see this guide for more details:
    # https://learn.hashicorp.com/consul/security-networking/certificates
    verify: true

    # If httpsOnly is true, Consul will disable the HTTP port on both
    # clients and servers and only accept HTTPS connections.
    httpsOnly: true

# Server, when enabled, configures a server cluster to run. This should
# be disabled if you plan on connecting to a Consul cluster external to
# the Kube cluster.
server:
  enabled: "-"
  image: null
  replicas: 3
  bootstrapExpect: 3 # Should <= replicas count

  # enterpriseLicense refers to a Kubernetes secret that you have created that
  # contains your enterprise license. It is required if you are using an
  # enterprise binary. Defining it here applies it to your cluster once a leader
  # has been elected. If you are not using an enterprise image
  # or if you plan to introduce the license key via another route, then set
  # these fields to null.
  enterpriseLicense:
    secretName: null
    secretKey: null

  # storage and storageClass are the settings for configuring stateful
  # storage for the server pods. storage should be set to the disk size of
  # the attached volume. storageClass is the class of storage which defaults
  # to null (the Kube cluster will pick the default).
  storage: 10Gi
  storageClass: null

  # connect will enable Connect on all the servers, initializing a CA
  # for Connect-related connections. Other customizations can be done
  # via the extraConfig setting.
  connect: true

  # Resource requests, limits, etc. for the server cluster placement. This
  # should map directly to the value of the resources field for a PodSpec,
  # formatted as a multi-line string. By default no direct resource request
  # is made.
  resources: null

  # updatePartition is used to control a careful rolling update of Consul
  # servers. This should be done particularly when changing the version
  # of Consul. Please refer to the documentation for more information.
  updatePartition: 0

  # disruptionBudget enables the creation of a PodDisruptionBudget to
  # prevent voluntary degrading of the Consul server cluster.
  disruptionBudget:
    enabled: true

    # maxUnavailable will default to (n/2)-1 where n is the number of
    # replicas. If you'd like a custom value, you can specify an override here.
    maxUnavailable: null

  # extraConfig is a raw string of extra configuration to set with the
  # server. This should be JSON.
  extraConfig: |
    {
      "ui": true,
      "client_addr": "0.0.0.0",
      "ca_file": "/consul/userconfig/consul-tls-server/ca.pem",
      "cert_file": "/consul/userconfig/consul-tls-server/consul.pem",
      "key_file": "/consul/userconfig/consul-tls-server/consul-key.pem",
      "ports": {
        "http": -1,
        "https": 8500
      }
    }

  # extraVolumes is a list of extra volumes to mount. These will be exposed
  # to Consul in the path `/consul/userconfig/<name>/`. The value below is
  # an array of objects, examples are shown below.
  extraVolumes:
    - type: secret
      name: consul-tls-server
      load: false # if true, will add to `-config-dir` to load by Consul

  # Affinity Settings
  # Commenting out or setting as empty the affinity variable, will allow
  # deployment to single node services such as Minikube
  affinity: |
    podAntiAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        - labelSelector:
            matchLabels:
              app: {{ template "consul.name" . }}
              release: "{{ .Release.Name }}"
              component: server
          topologyKey: kubernetes.io/hostname

  # Toleration Settings for server pods
  # This should be a multi-line string matching the Toleration array
  # in a PodSpec.
  tolerations: ""

  # nodeSelector labels for server pod assignment, formatted as a muli-line string.
  # ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
  # Example:
  # nodeSelector: |
  #   beta.kubernetes.io/arch: amd64
  nodeSelector: null

  # used to assign priority to server pods
  # ref: https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/
  priorityClassName: ""

  # Extra annotations to attach to the server pods
  # This should be a multi-line string mapping directly to the a map of
  # the annotations to apply to the server pods
  annotations: null

  # extraEnvVars is a list of extra enviroment variables to set with the stateful set. These could be
  # used to include proxy settings required for cloud auto-join feature,
  # in case kubernetes cluster is behind egress http proxies. Additionally, it could be used to configure
  # custom consul parameters.
  extraEnvironmentVars:
    CONSUL_CACERT: /consul/userconfig/consul-tls-server/ca.pem
    CONSUL_CLIENT_CERT: /consul/userconfig/consul-tls-server/consul.pem
    CONSUL_CLIENT_KEY: /consul/userconfig/consul-tls-server/consul-key.pem
    CONSUL_HTTP_SSL: true
    # http_proxy: http://localhost:3128,
    # https_proxy: http://localhost:3128,
    # no_proxy: internal.domain.com

# Client, when enabled, configures Consul clients to run on every node
# within the Kube cluster. The current deployment model follows a traditional
# DC where a single agent is deployed per node.
client:
  enabled: true
  image: null
  join: null


  dataDirectoryHostPath: null

  # If true, Consul's gRPC port will be exposed (see https://www.consul.io/docs/agent/options.html#grpc_port).
  # This should be set to true if connectInject or meshGateway is enabled.
  grpc: true

  exposeGossipPorts: false

  # Resource requests, limits, etc. for the client cluster placement. This
  # should map directly to the value of the resources field for a PodSpec,
  # formatted as a multi-line string. By default no direct resource request
  # is made.
  resources: null

  # extraConfig is a raw string of extra configuration to set with the
  # client. This should be JSON.
  extraConfig: |
    {
      "ca_file": "/consul/userconfig/consul-tls-client/ca.pem",
      "cert_file": "/consul/userconfig/consul-tls-client/consul.pem",
      "key_file": "/consul/userconfig/consul-tls-client/consul-key.pem",
      "ports": {
        "http": -1,
        "https": 8500
      }
    }

  # extraVolumes is a list of extra volumes to mount. These will be exposed
  # to Consul in the path `/consul/userconfig/<name>/`. The value below is
  # an array of objects, examples are shown below.
  extraVolumes:
    - type: secret 
      name: consul-tls-client
      load: false # if true, will add to `-config-dir` to load by Consul

  # Toleration Settings for Client pods
  # This should be a multi-line string matching the Toleration array
  # in a PodSpec.
  # The example below will allow Client pods to run on every node
  # regardless of taints
  # tolerations: |
  #   - operator: "Exists"
  tolerations: ""

  # nodeSelector labels for client pod assignment, formatted as a multi-line string.
  # ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
  # Example:
  # nodeSelector: |
  #   beta.kubernetes.io/arch: amd64
  nodeSelector: null

  # Affinity Settings for Client pods, formatted as a multi-line YAML string.
  # ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
  # Example:
  # affinity: |
  #   nodeAffinity:
  #     requiredDuringSchedulingIgnoredDuringExecution:
  #       nodeSelectorTerms:
  #       - matchExpressions:
  #         - key: node-role.kubernetes.io/master
  #           operator: DoesNotExist
  affinity: {}

  # used to assign priority to client pods
  # ref: https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/
  priorityClassName: ""

  # Extra annotations to attach to the client pods
  # This should be a multi-line string mapping directly to the a map of
  # the annotations to apply to the client pods
  annotations: null

  # extraEnvVars is a list of extra enviroment variables to set with the pod. These could be
  # used to include proxy settings required for cloud auto-join feature,
  # in case kubernetes cluster is behind egress http proxies. Additionally, it could be used to configure
  # custom consul parameters.
  extraEnvironmentVars:
    CONSUL_CACERT: /consul/userconfig/consul-tls-client/ca.pem
    CONSUL_CLIENT_CERT: /consul/userconfig/consul-tls-client/consul.pem
    CONSUL_CLIENT_KEY: /consul/userconfig/consul-tls-client/consul-key.pem
    CONSUL_HTTP_SSL: true
    # http_proxy: http://localhost:3128,
    # https_proxy: http://localhost:3128,
    # no_proxy: internal.domain.com

  # dnsPolicy to use.
  dnsPolicy: null

  # updateStrategy for the DaemonSet.
  # See https://kubernetes.io/docs/tasks/manage-daemon/update-daemon-set/#daemonset-update-strategy.
  # This should be a multi-line string mapping directly to the updateStrategy
  # Example:
  #  updateStrategy: |
  #    rollingUpdate:
  #      maxUnavailable: 5
  #    type: RollingUpdate
  updateStrategy: null

  # snaphotAgent contains settings for setting up and running snapshot agents
  # within the Consul clusters. They are required to be co-located with Consul
  # clients, so will inherit the clients' nodeSelector, tolerations and affinity.
  # This is an Enterprise feature only.
  snapshotAgent:
    enabled: false

    # replicas determines how many snapshot agent pods are created
    replicas: 2

    # configSecret references a Kubernetes secret that should be manually created to
    # contain the entire config to be used on the snapshot agent. This is the preferred
    # method of configuration since there are usually storage credentials present.
    # Snapshot agent config details:
    # https://www.consul.io/docs/commands/snapshot/agent.html#config-file-options-
    # To create a secret:
    # https://kubernetes.io/docs/concepts/configuration/secret/#creating-a-secret-using-kubectl-create-secret
    configSecret:
      secretName: null
      secretKey: null

# Configuration for DNS configuration within the Kubernetes cluster.
# This creates a service that routes to all agents (client or server)
# for serving DNS requests. This DOES NOT automatically configure kube-dns
# today, so you must still manually configure a `stubDomain` with kube-dns
# for this to have any effect:
# https://kubernetes.io/docs/tasks/administer-cluster/dns-custom-nameservers/#configure-stub-domain-and-upstream-dns-servers
dns:
  enabled: "-"

  # Set a predefined cluster IP for the DNS service.
  # Useful if you need to reference the DNS service's IP
  # address in CoreDNS config.
  clusterIP: null

  # Extra annotations to attach to the dns service
  # This should be a multi-line string of
  # annotations to apply to the dns Service
  annotations: null

ui:
  # True if you want to enable the Consul UI. The UI will run only
  # on the server nodes. This makes UI access via the service below (if
  # enabled) predictable rather than "any node" if you're running Consul
  # clients as well.
  enabled: "-"

  # True if you want to create a Service entry for the Consul UI.
  #
  # serviceType can be used to control the type of service created. For
  # example, setting this to "LoadBalancer" will create an external load
  # balancer (for supported K8S installations) to access the UI.
  service:
    enabled: true
    type: null
    # This should be a multi-line string mapping directly to the a map of
    # the annotations to apply to the UI service
    annotations: null
    # Additional ServiceSpec values
    # This should be a multi-line string mapping directly to a Kubernetes
    # ServiceSpec object.
    additionalSpec: null

# syncCatalog will run the catalog sync process to sync K8S with Consul
# services. This can run bidirectional (default) or unidirectionally (Consul
# to K8S or K8S to Consul only).
#
# This process assumes that a Consul agent is available on the host IP.
# This is done automatically if clients are enabled. If clients are not
# enabled then set the node selection so that it chooses a node with a
# Consul agent.
syncCatalog:
  # True if you want to enable the catalog sync. Set to "-" to inherit from
  # global.enabled.
  enabled: false
  image: null
  default: true # true will sync by default, otherwise requires annotation

  # toConsul and toK8S control whether syncing is enabled to Consul or K8S
  # as a destination. If both of these are disabled, the sync will do nothing.
  toConsul: true
  toK8S: true

  # k8sPrefix is the service prefix to prepend to services before registering
  # with Kubernetes. For example "consul-" will register all services
  # prepended with "consul-". (Consul -> Kubernetes sync)
  k8sPrefix: null

  # k8sSourceNamespace is the Kubernetes namespace to watch for service
  # changes and sync to Consul. If this is not set then it will default
  # to all namespaces.
  k8sSourceNamespace: null


  addK8SNamespaceSuffix: true


  # of existing services in Consul and registering them with a new name.
  consulPrefix: null

  # k8sTag is an optional tag that is applied to all of the Kubernetes services
  # that are synced into Consul. If nothing is set, defaults to "k8s".
  # (Kubernetes -> Consul sync)
  k8sTag: null

  # syncClusterIPServices syncs services of the ClusterIP type, which may
  # or may not be broadly accessible depending on your Kubernetes cluster.
  # Set this to false to skip syncing ClusterIP services.
  syncClusterIPServices: true


  nodePortSyncType: ExternalFirst

  # aclSyncToken refers to a Kubernetes secret that you have created that contains
  # an ACL token for your Consul cluster which allows the sync process the correct
  # permissions. This is only needed if ACLs are enabled on the Consul cluster.
  aclSyncToken:
    secretName: null
    secretKey: null

  # nodeSelector labels for syncCatalog pod assignment, formatted as a multi-line string.
  # ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
  # Example:
  # nodeSelector: |
  #   beta.kubernetes.io/arch: amd64
  nodeSelector: null

  # Log verbosity level. One of "trace", "debug", "info", "warn", or "error".
  logLevel: info

  # Override the default interval to perform syncing operations creating Consul services.
  consulWriteInterval: null

# ConnectInject will enable the automatic Connect sidecar injector.
connectInject:
  # True if you want to enable connect injection. Set to "-" to inherit from
  # global.enabled.
  # Requires consul-k8s >= 0.10.1.
  enabled: false
  image: null # image for consul-k8s that contains the injector
  default: false # true will inject by default, otherwise requires annotation

  imageConsul: null
  imageEnvoy: null


  namespaceSelector: null


  certs:

    secretName: null

    # caBundle is a base64-encoded PEM-encoded certificate bundle for the
    # CA that signed the TLS certificate that the webhook serves. This must
    # be set if secretName is non-null.
    caBundle: ""

    certName: tls.crt
    keyName: tls.key

  nodeSelector: null


  aclBindingRuleSelector: "serviceaccount.name!=default"

  # If not using global.bootstrapACLs and instead manually setting up an auth
  # method for Connect inject, set this to the name of your auth method.
  overrideAuthMethodName: ""

  # Requires Consul >= v1.5 and consul-k8s >= v0.8.1.
  centralConfig:

    enabled: true

    defaultProtocol: null

    proxyDefaults: |
      {}

# Mesh Gateways enable Consul Connect to work across Consul datacenters.
meshGateway:

  enabled: false

  globalMode: local

  # Number of replicas for the Deployment.
  replicas: 2

  # What gets registered as wan address for the gateway.
  wanAddress:
    # Port that gets registered.
    port: 443

    useNodeIP: true


    useNodeName: false

    host: ""

  # The service option configures the Service that fronts the Gateway Deployment.
  service:
    # Whether to create a Service or not.
    enabled: false

    # Type of service, ex. LoadBalancer, ClusterIP.
    type: ClusterIP

    # Port that the service will be exposed on.
    # The targetPort will be set to meshGateway.containerPort.
    port: 443

    # Optional nodePort of the service. Can be used in conjunction with
    # type: NodePort.
    nodePort: null

    # Optional YAML string for additional annotations.
    annotations: null

    # Optional YAML string that will be appended to the Service spec.
    additionalSpec: null

  # Envoy image to use.
  imageEnvoy: envoyproxy/envoy:v1.10.0

  # If set to true, gateway Pods will run on the host network.
  hostNetwork: false

  # dnsPolicy to use.
  dnsPolicy: null


  consulServiceName: ""

  # Port that the gateway will run on inside the container.
  containerPort: 443

  hostPort: null

  enableHealthChecks: true

  resources: |
    requests:
      memory: "128Mi"
      cpu: "100m"
    limits:
      memory: "256Mi"
      cpu: "150m"

  affinity: |
    podAntiAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        - labelSelector:
            matchLabels:
              app: {{ template "consul.name" . }}
              release: "{{ .Release.Name }}"
              component: mesh-gateway
          topologyKey: kubernetes.io/hostname

  tolerations: null

  nodeSelector: null

  priorityClassName: ""

  annotations: null


tests:
  enabled: false
```

LOGS:
```
    2020/01/22 16:35:15 [INFO] agent: (LAN) joining: [prod-dc-consul-server-0.prod-dc-consul-server.vault-consul.svc prod-dc-consul-server-1.prod-dc-consul-server.vault-consul.svc prod-dc-consul-server-2.prod-dc-consul-server.vault-consul.svc]
    2020/01/22 16:35:15 [INFO] agent: started state syncer
==> Consul agent running!
    2020/01/22 16:35:15 [WARN] memberlist: Failed to resolve prod-dc-consul-server-0.prod-dc-consul-server.vault-consul.svc: lookup prod-dc-consul-server-0.prod-dc-consul-server.vault-consul.svc on 10.31.240.10:53: no such host
    2020/01/22 16:35:15 [WARN] memberlist: Failed to resolve prod-dc-consul-server-1.prod-dc-consul-server.vault-consul.svc: lookup prod-dc-consul-server-1.prod-dc-consul-server.vault-consul.svc on 10.31.240.10:53: no such host
    2020/01/22 16:35:15 [WARN] memberlist: Failed to resolve prod-dc-consul-server-2.prod-dc-consul-server.vault-consul.svc: lookup prod-dc-consul-server-2.prod-dc-consul-server.vault-consul.svc on 10.31.240.10:53: no such host
    2020/01/22 16:35:15 [WARN] agent: (LAN) couldn't join: 0 Err: 3 errors occurred:
	* Failed to resolve prod-dc-consul-server-0.prod-dc-consul-server.vault-consul.svc: lookup prod-dc-consul-server-0.prod-dc-consul-server.vault-consul.svc on 10.31.240.10:53: no such host
	* Failed to resolve prod-dc-consul-server-1.prod-dc-consul-server.vault-consul.svc: lookup prod-dc-consul-server-1.prod-dc-consul-server.vault-consul.svc on 10.31.240.10:53: no such host
	* Failed to resolve prod-dc-consul-server-2.prod-dc-consul-server.vault-consul.svc: lookup prod-dc-consul-server-2.prod-dc-consul-server.vault-consul.svc on 10.31.240.10:53: no such host

```

**Comment by lkysow** [comment]
Hi @wiltonfelix please open a new issue. This ticket is for "no cluster leader" which doesn't occur in your logs. Also please only include the values you're overriding, not the whole values.yaml, there's so much in there it's too hard to figure out what your specific config is.

I'm closing this issue as @yueru-mylove  never responded.
